File: java-service/src/main/java/com/analyticore/analyseservice/domain/port/JobRepositoryPort.java
package com.analyticore.analyseservice.domain.port;

import com.analyticore.analyseservice.domain.Job;
import java.util.Optional;
import java.util.UUID;

/**
 * Puerto de dominio — contrato de persistencia de jobs (inversión de dependencias).
 * La capa de aplicación depende de esta interfaz, no de JPA ni PostgreSQL.
 */
public interface JobRepositoryPort {

    Optional<Job> findById(UUID jobId);

    Job save(Job job);

    Optional<Job> findTopByOrderByCreatedAtDesc();
}

File: python-service/domain/ports/job_repository_port.py
"""Puerto de dominio — contrato de persistencia de jobs (inversión de dependencias)."""
from abc import ABC, abstractmethod
from uuid import UUID

from domain.models import Job


class JobRepositoryPort(ABC):
    @abstractmethod
    def save(self, job: Job) -> Job:
        pass

    @abstractmethod
    def find_by_id(self, job_id: UUID) -> Job | None:
        pass

    @abstractmethod
    def find_latest(self) -> Job | None:
        pass
